Add aws-sdk-cpp recipe

jakebolewski opened the pull request.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/aws-sdk-cpp```) and found it was in an excellent condition.

ocefpaf commented on recipes/aws-sdk-cpp/build.sh:
```diff
@@ -0,0 +1,9 @@
+  -DENABLE_TESTING=OFF \
+  -DCMAKE_BUILD_TYPE=Release ..
+make -j${CPU_COUNT}
+make install
```
Is there a `make check` we can run here?

ocefpaf commented on recipes/aws-sdk-cpp/meta.yaml:
```diff
@@ -0,0 +1,58 @@
+    - toolchain
+    - cmake >=3.2
+    - curl
+    - openssl
```
Can you please add the pinning for these two pinned here and below?

```
- curl >=7.44.0,<8
- openssl 1.0.*
```

PS: note that we are in the process to change how we deal with pinned dependencies,
but we move to conda-build 3 we still need to add the pins here.

ocefpaf commented on recipes/aws-sdk-cpp/meta.yaml:
```diff
@@ -0,0 +1,58 @@
+test:
+  requires:
+    - gcc  # [osx]
+
```
No need for this space.

jakebolewski commented:
Hi @ocefpaf  thanks for taking the time to review!

I believe I addressed all your comments.  I re-enabled testing, there is no `make check` in the SDK so the tests are run after each component is built.

isuruf commented on recipes/aws-sdk-cpp/run_test.sh:
```diff
@@ -0,0 +1,14 @@
+
+if [ "$(uname)" == "Darwin" ]; then
+    # OSX
+    g++ test.cpp -I"${PREFIX}/include" -L"${PREFIX}/lib" -laws-cpp-sdk-core -o test-core
```
Change this `clang++ test.cpp -I"${PREFIX}/include" -L"${PREFIX}/lib" -Wl,-rpath,${PREFIX}/lib -laws-cpp-sdk-core -o test-core`

isuruf commented on recipes/aws-sdk-cpp/run_test.sh:
```diff
@@ -0,0 +1,14 @@
+if [ "$(uname)" == "Darwin" ]; then
+    # OSX
+    g++ test.cpp -I"${PREFIX}/include" -L"${PREFIX}/lib" -laws-cpp-sdk-core -o test-core
+    env DYLD_LIBRARY_PATH="${PREFIX}/lib:$DYLD_LIBRARY_PATH" ./test-core
```
Remove `env DYLD_LIBRARY_PATH="${PREFIX}/lib:$DYLD_LIBRARY_PATH"` after adding rpath as above

isuruf commented on recipes/aws-sdk-cpp/run_test.sh:
```diff
@@ -0,0 +1,14 @@
+else
+    # Linux
+    g++ test.cpp -I"${PREFIX}/include" -L"${PREFIX}/lib${ARCH}" -laws-cpp-sdk-core -o test-core
+    env LD_LIBRARY_PATH="${PREFIX}/lib${ARCH}:$LD_LIBRARY_PATH" ./test-core
```
Add rpath as above and remove `LD_LIBRARY_PATH`

isuruf commented on recipes/aws-sdk-cpp/meta.yaml:
```diff
@@ -0,0 +1,55 @@
+
+test:
+  requires:
+    - gcc  # [osx]
```
Remove this

jakebolewski commented:
@isuruf thanks, updated

isuruf commented on recipes/aws-sdk-cpp/run_test.sh:
```diff
@@ -0,0 +1,12 @@
+    clang++ test.cpp -I"${PREFIX}/include" -L"${PREFIX}/lib" -Wl,-rpath,"${PREFIX}/lib" -laws-cpp-sdk-core -o test-core
+else
+    # Linux
+    g++ test.cpp -I"${PREFIX}/include" -L"${PREFIX}/lib${ARCH}" -Wl,-rpath,"${PREFIX}/lib${ARCH}" -laws-cpp-sdk-core -o test-core
```
Add `-DCMAKE_INSTALL_LIBDIR=lib` when building and remove `${ARCH}`. We are using `${PREFIX}/lib` for all packages.

jakebolewski replied:
ok, updated

isuruf commented on recipes/aws-sdk-cpp/run_test.sh:
```diff
@@ -0,0 +1,12 @@
+    clang++ test.cpp -I"${PREFIX}/include" -L"${PREFIX}/lib" -Wl,-rpath,"${PREFIX}/lib" -laws-cpp-sdk-core -o test-core
+else
+    # Linux
+    g++ test.cpp -I"${PREFIX}/include" -L"${PREFIX}/lib" -Wl,-rpath,"${PREFIX}/lib$" -laws-cpp-sdk-core -o test-core
```
Typo. `lib$`->`lib`

jakebolewski replied:
ghaa, thanks again :)

isuruf commented on recipes/aws-sdk-cpp/run_test.sh:
```diff
@@ -0,0 +1,12 @@
+else
+    # Linux
+    g++ test.cpp -I"${PREFIX}/include" -L"${PREFIX}/lib" -Wl,-rpath,"${PREFIX}/lib" -laws-cpp-sdk-core -o test-core
+fi
```
Why did you remove running the executable?

jakebolewski replied:
sorry, just trying to wrap this up quickly before leaving work

added it back

ocefpaf commented:
Thanks @isuruf for looking into this!
Thanks @jakebolewski for the PR!
